api/desecapi/utils.py

```diff
@@ -0,0 +1,14 @@
+from django.db.models import CharField, F, Value
+from django.db.models.functions import Concat, Length
+
+from desecapi.models import Domain
+
+
+def get_domain_for_qname(qname, **kwargs):
```

Comment:
1. this function deserves extensive tests
2. this could live in the `Domain` class instead of creating this new module
3. add type annotations?

Reply:
done